src/coord/src/coord/dataflow_builder.rs

```diff
@@ -115,6 +120,35 @@ impl<'a> DataflowBuilder<'a> {
                         );
                     }
                     CatalogItem::Source(source) => {
+                        if source.connector.requires_single_materialization() {
+                            let source_type = match &source.connector {
```

Comment:
can we avoid referring to specific connector types here? If we need to extract a particular piece of info we could add a method and make it part of the connector interface like the `requires_single_materialization` call like `.detailed_name()` or something like that

Reply:
I can move the logic out of here and onto the `RematerializedSourceType` itself, but it's an error display helper that is used for a couple of things (linking to specific documentation pages and in the error message). Using a string that is derived from the SourceConnector wouldn't really work. It would require either matching on the string in the error display or making the error not as helpful. Or, I guess, a brand new docs page.

I've moved things around, let me know what you think of it.